Xamarin.PropertyEditing.Mac/Controls/BindingEditor/CreateValueConverterWindow.cs

```diff
@@ -0,0 +1,125 @@
+				NSLayoutConstraint.Create (valueConverterLabel, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, 24),
+			});
+
+			this.valueConverterName = new NSTextField {
+				ControlSize = NSControlSize.Small,
+				TranslatesAutoresizingMaskIntoConstraints = false,
+			};
+			container.AddSubview (this.valueConverterName);
+
+			container.AddConstraints (new[] {
+				NSLayoutConstraint.Create (this.valueConverterName, NSLayoutAttribute.Top, NSLayoutRelation.Equal, valueConverterLabel, NSLayoutAttribute.Bottom, 1f, 1f),
+				NSLayoutConstraint.Create (this.valueConverterName, NSLayoutAttribute.Left, NSLayoutRelation.Equal, container, NSLayoutAttribute.Left, 1f, 5f),
+				NSLayoutConstraint.Create (this.valueConverterName, NSLayoutAttribute.Width, NSLayoutRelation.Equal, container, NSLayoutAttribute.Width, 1f, -10f),
+				NSLayoutConstraint.Create (this.valueConverterName, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, 24),
+			});
+
+			var typeSelectorControl = new TypeSelectorControl {
+				ViewModel = ViewModel,
+				TranslatesAutoresizingMaskIntoConstraints = false,
+			};
+
+			container.AddSubview (typeSelectorControl);
+
+			container.AddConstraints (new[] {
+				NSLayoutConstraint.Create (typeSelectorControl, NSLayoutAttribute.Top, NSLayoutRelation.Equal, container, NSLayoutAttribute.Top, 1f, 45f),
+				NSLayoutConstraint.Create (typeSelectorControl, NSLayoutAttribute.Left, NSLayoutRelation.Equal, container, NSLayoutAttribute.Left, 1f, 0f),
+				NSLayoutConstraint.Create (typeSelectorControl, NSLayoutAttribute.Width, NSLayoutRelation.Equal, container, NSLayoutAttribute.Width, 1f, 0f),
+				NSLayoutConstraint.Create (typeSelectorControl, NSLayoutAttribute.Height, NSLayoutRelation.Equal, container, NSLayoutAttribute.Height, 1f, -50f)
+			});
+
+			var buttonDone = new NSButton {
+				BezelStyle = NSBezelStyle.Rounded,
+				Highlighted = true,
+				KeyEquivalent = "\r", // Fire when enter pressed
+				Title = Properties.Resources.DoneTitle,
+				TranslatesAutoresizingMaskIntoConstraints = false,
+			};
+
+			buttonDone.Activated += (sender, e) => {
+				Delegate.Response = NSModalResponse.OK;
+				Close ();
+			};
+
+			container.AddSubview (buttonDone);
+
+			container.AddConstraints (new[] {
+				NSLayoutConstraint.Create (buttonDone, NSLayoutAttribute.Top, NSLayoutRelation.Equal, container, NSLayoutAttribute.Bottom, 1f, -32f),
+				NSLayoutConstraint.Create (buttonDone, NSLayoutAttribute.Right, NSLayoutRelation.Equal, container, NSLayoutAttribute.Right, 1f, -16f),
+				NSLayoutConstraint.Create (buttonDone, NSLayoutAttribute.Height, NSLayoutRelation.Equal, 1f, 24),
+			});
+
+			ContentViewController = new NSViewController (null, null) {
+				View = container,
+			};
+
+			ViewModel.PropertyChanged += (sender, e) => {
+				if (e.PropertyName == nameof (ViewModel.SelectedType)) {
+					this.valueConverterName.StringValue = ViewModel.SelectedType.Name;
```

Comment:
This will be null if you select something in the treeview and then click on an empty space to unselect it.

Reply:
This was marked resolved but wasn't actually fixed.

Reply:
Not sure how I lost this change. Will be fixed in my next push.